lib/identity_cache/query_api.rb

```diff
@@ -179,9 +175,43 @@ def require_if_necessary #:nodoc:
       end
 
       def resolve_cache_miss(id)
-        object = self.includes(cache_fetch_includes).reorder(nil).where(primary_key => id).first
-        object.send(:populate_association_caches) if object
-        object
+        record = self.includes(cache_fetch_includes).reorder(nil).where(primary_key => id).first
+        preload_id_embedded_associations([record]) if record
+        record
+      end
+
+      def preload_id_embedded_associations(records)
+        return if records.empty?
+        each_id_embedded_association do |name, options|
```

Comment:
Is `name` being used here? Does it make sense to pass it if it is not being used?